File: dotnet-resend-examples/Examples/ScheduledSend.cs
namespace ResendExamples;

using Resend;

public static class ScheduledSend
{
    public static async Task RunAsync()
    {
        var apiKey = Environment.GetEnvironmentVariable("RESEND_API_KEY")
            ?? throw new Exception("RESEND_API_KEY environment variable is required");

        var client = new ResendClient(apiKey);

        var from = Environment.GetEnvironmentVariable("EMAIL_FROM") ?? "Acme <onboarding@resend.dev>";

        // Schedule for 5 minutes in the future (maximum: 7 days)
        var scheduledAt = DateTime.UtcNow.AddMinutes(5);

        var message = new EmailMessage
        {
            From = from,
            To = { "delivered@resend.dev" },
            Subject = "Scheduled Email from .NET",
            HtmlBody = "<h1>Hello from the future!</h1><p>This email was scheduled for later delivery.</p>",
            TextBody = "Hello from the future! This email was scheduled for later delivery.",
            ScheduledAt = scheduledAt
        };

        var response = await client.EmailSendAsync(message);

        Console.WriteLine("Email scheduled successfully!");
        Console.WriteLine($"Email ID: {response.Id}");
        Console.WriteLine($"Scheduled for: {scheduledAt:yyyy-MM-dd HH:mm:ss} UTC");
    }
}

File: rust-resend-examples/examples/scheduled_send.rs
use chrono::Utc;
use resend_rs::types::CreateEmailBaseOptions;
use resend_rs::Resend;

#[tokio::main]
async fn main() {
    dotenvy::dotenv().ok();

    let api_key = std::env::var("RESEND_API_KEY").expect("RESEND_API_KEY environment variable is required");
    let from = std::env::var("EMAIL_FROM").unwrap_or_else(|_| "Acme <onboarding@resend.dev>".to_string());

    let resend = Resend::new(&api_key);

    // Schedule for 5 minutes in the future (maximum: 7 days)
    let scheduled_at = Utc::now() + chrono::Duration::minutes(5);
    let scheduled_at_iso = scheduled_at.to_rfc3339();

    let email = CreateEmailBaseOptions::new(&from, ["delivered@resend.dev"], "Scheduled Email from Rust")
        .with_html("<h1>Hello from the future!</h1><p>This email was scheduled for later delivery.</p>")
        .with_text("Hello from the future! This email was scheduled for later delivery.")
        .with_scheduled_at(&scheduled_at_iso);

    match resend.emails.send(email).await {
        Ok(response) => {
            println!("Email scheduled successfully!");
            println!("Email ID: {}", response.id);
            println!("Scheduled for: {} UTC", scheduled_at.format("%Y-%m-%d %H:%M:%S"));
        }
        Err(e) => {
            eprintln!("Error scheduling email: {}", e);
            std::process::exit(1);
        }
    }
}
